go/visu/experiment/experiment.go

```diff
@@ -96,18 +96,18 @@ func BuildMeta(folder string) (*Meta, error) {
 	}
 
 	for !parser.IsFinished() {
-		meta.SamplesDuration = append(meta.SamplesDuration, parser.Uint32())
+		meta.Samples[parser.String()] = parser.Uint32()
 	}
 
 	return meta, nil
 }
 
-func BuildData(meta *Meta, sampleI int) (*Data, error) {
-	if sampleI >= len(meta.SamplesDuration) {
-		return nil, errors.New("sampleI is too big")
+func BuildData(meta *Meta, sampleName string) (*Data, error) {
+	duration, present := meta.Samples[sampleName]
+	if !present {
+		return nil, errors.New(fmt.Sprintf("Sample %v is not present", sampleName))
 	}
-	duration := meta.SamplesDuration[sampleI]
-	fileName := fmt.Sprintf("data_%d.bin", sampleI)
+	fileName := fmt.Sprintf("data_%v.bin", sampleName)
```

Comment:
So I don't get something, in the end the plots have the internal .bin in the name, why?

Reply:
fix this